cpp/include/cudf/transform.hpp

```diff
@@ -32,27 +32,32 @@ namespace CUDF_EXPORT cudf {
  */
 
 /**
- * @brief Creates a new column by applying a unary function against every
- * element of an input column.
+ * @brief Creates a new column by applying a transform function against every
+ * element of the input columns.
  *
  * Computes:
- * `out[i] = F(in[i])`
+ * `out[i] = F(inputs[i]...)`.
  *
- * The output null mask is the same is the input null mask so if input[i] is
- * null then output[i] is also null
+ * Note that for every scalar in `inputs` (columns of size 1), `input[i] == input[0]`
```

Comment:
Is it worth making this work for true scalars, rather than length 1 columns? The current behavior is unexpected to me, at least on first read. I didn't see this comment until after I had read through the rest of the PR and I was getting confused about how things like the indexing were working when one element was a scalar, then I realized it wasn't a true scalar.

Reply:
How do you propose that would work?
The other alternative would be to take a variant of columns and true scalars.

Reply:
Cython doesn't support variants so that would be tricky

Reply:
I'm ok with using a single-row column to represent a scalar. 
We've had the scalar/column variant issue come up before on other APIs like concatenate and in some internal code logic but I don't think it is worth solving in this code or holding up this PR.

Reply:
Yeah, it feels wrong to me too, but due to the nature of transforms, I don't think it is a wrong design.
Also, we don't have scalars that are functionally equivalent to Columns in the C++ side, The scalars we have are type-specific and would be difficult to work with. Alternatively, we could create another scalar class that just wraps the column class.